### PR TITLE
core, cmd: fill blockNumber in logs

### PR DESCRIPTION
**Pull request by rjl493456442**

### CODE REVIEW

File: core/state/statedb.go [hunk 1]
```diff
@@ -206,9 +206,10 @@ func (s *StateDB) AddLog(log *types.Log) {
 	s.logSize++
 }
 
-func (s *StateDB) GetLogs(hash common.Hash, blockHash common.Hash) []*types.Log {
+func (s *StateDB) GetLogs(hash common.Hash, blockNumber uint64, blockHash common.Hash) []*types.Log {
```
Comment (holiman) [inline core/state/statedb.go:209]: Please add a docstring , e.g. 
```suggestion
// GetLogs returns the logs matching the specified transaction hash, and annotates them with the given blocknumber/blockhash 
func (s *StateDB) GetLogs(hash common.Hash, blockNumber uint64, blockHash common.Hash) []*types.Log {
```

**Review by holiman — APPROVED** [review]
LGTM (minor nit)

**Review by s1na — APPROVED** [review]
LGTM

File: core/types/log.go [hunk 2]
```diff
@@ -41,12 +41,12 @@ type Log struct {
...
 	// block in which the transaction was included
 	BlockNumber uint64 `json:"blockNumber"`
+	// hash of the block in which the transaction was included
+	BlockHash common.Hash `json:"blockHash"`
 	// hash of the transaction
 	TxHash common.Hash `json:"transactionHash" gencodec:"required"`
 	// index of the transaction in the block
 	TxIndex uint `json:"transactionIndex"`
-	// hash of the block in which the transaction was included
-	BlockHash common.Hash `json:"blockHash"`
```
[the remark this answers is not in the document]
Comment (rjl493456442) [reply core/types/log.go]: Reverted. But any particular reason that we can't reorder fields?

**Comment by holiman** [comment]
Sorry, something about my comment is not approved by the linter, for some reason


```
core/state/statedb.go:209: File is not `goimports`-ed (goimports)
// GetLogs returns the logs matching the specified transaction hash, and annotates them with the given blocknumber/blockhash
```

**Pull request merged**